dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/osgi/WebContainer.java

```diff
@@ -1568,41 +1580,56 @@ protected void unsetCacheServletWrapperFactory(CacheServletWrapperFactory factor
         // no-op intended here to avoid cacheServletWrapperFactory being null when switching service implementations
     }
 
-    
-    @Reference(service=ServletVersion.class, cardinality=ReferenceCardinality.MANDATORY, policy=ReferencePolicy.DYNAMIC, policyOption=ReferencePolicyOption.GREEDY)
-    protected synchronized void setVersion(ServiceReference<ServletVersion> reference) {
+   /*
+    * Should only be called by loadServletSpecVersion
+    */
+    public static void setVersion(int version){
         String methodName = "setVersion";
-        versionRef = reference;
-        WebContainer.loadedContainerSpecLevel = (Integer) reference.getProperty("version");
+        WebContainer.loadedContainerSpecLevel = version;
         if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
             Tr.debug(tc, methodName, "loadedContainerSpecLevel [ " + WebContainer.loadedContainerSpecLevel + " ]");
         }
     }
 
-    protected synchronized void unsetVersion(ServiceReference<ServletVersion> reference) {
-        if (reference == this.versionRef) {
-            versionRef = null;
-            WebContainer.loadedContainerSpecLevel = DEFAULT_MAX_VERSION;
-        }
+   /*
+    * Not sure if this should exist or if it should be called when webcontainer is deactivating? 
+    */
+    public static void unsetVersion(int version){
+        WebContainer.loadedContainerSpecLevel = DEFAULT_MAX_VERSION;
     }
```

Comment:
Let you know if I should shall `unsetVersion` or not?  When a new servlet feature is loaded, the older classes are discarded, so a new version will be set.  Because of this, it seems unecessary to call unsetVersion?

Reply:
I don't see where this unsetter has been called either.  We can just leave it there but prefer to change to "protected" instead of static